SHA512/256 build fail   on arm64 PC

### Summary
This issue is found in Arm64 PC. To run ` all.sh` in aarch64 ubuntu 20.04, it reports sha512/256 error.

- By default, `./tests/scripts/all.sh build_psa_accel_alg_sha*` will report compiler options error.
   - And I check `all.sh` , I did not found any  `aarch64-linux-gnu` cross compile test . I think the test coverage is not enough. we should add `aarch64-linux-gnu` cross compiler, and add tests for it.


- Test command: `./tests/scripts/all.sh build_no_std_function`, it reports below error. With `-march=armv8.2-a+sha3+crypto`
```
/home/haoyu01/work/mbedtls/mbedtls/library/sha512.c:530:5: error: no previous prototype for 'mbedtls_internal_sha512_process_a64_crypto' [-Werror=missing-prototypes]
  530 | int mbedtls_internal_sha512_process_a64_crypto( mbedtls_sha512_context *ctx,
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/haoyu01/work/mbedtls/mbedtls/library/sha512.c:548:5: error: no previous prototype for 'mbedtls_internal_sha512_process_c' [-Werror=missing-prototypes]
  548 | int mbedtls_internal_sha512_process_c( mbedtls_sha512_context *ctx,
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/haoyu01/work/mbedtls/mbedtls/library/sha512.c:679:5: error: no previous prototype for 'mbedtls_a64_crypto_sha512_has_support' [-Werror=missing-prototypes]
  679 | int mbedtls_a64_crypto_sha512_has_support( void )
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

- Test command : `./tests/scripts/all.sh check_names --force` report below errors. With `-march=armv8.2-a+sha3+crypto`
```
All symbols in header: FAIL

    > 'mbedtls_a64_crypto_sha512_has_support' was found as an available symbol
      in the output of nm, however it was not declared in any header files.
    > 'mbedtls_internal_sha512_process_a64_crypto' was found as an available
      symbol in the output of nm, however it was not declared in any header
      files.
    > 'mbedtls_internal_sha512_process_c' was found as an available symbol in
      the output of nm, however it was not declared in any header files.
```


### System information

Mbed TLS version (number or commit id): 09dc05b880b1b32972bf1052b558dbc0aaca92db
Operating system and version: ubuntu 20.04 
Machine ARCH: Arm64

### Expected behavior
Test PASS


### Actual behavior
Test Fail

### Additional information
To add `-march=armv8.2-a+sha3+crypto` with below patch.
```diff
diff --git a/include/mbedtls/build_info.h b/include/mbedtls/build_info.h
index 3dd944c76..8d4830d91 100644
--- a/include/mbedtls/build_info.h
+++ b/include/mbedtls/build_info.h
@@ -80,6 +80,9 @@
 #if defined(MBEDTLS_PSA_CRYPTO_CONFIG)
 #include "mbedtls/config_psa.h"
 #endif
+#if defined(__aarch64__)
+#pragma GCC target ("arch=armv8.2-a+sha3+crypto")
+#endif
 
 #include "mbedtls/check_config.h"
```

@yuhaoth I only get this error if I add `-Wmissing-prototypes`, which doesn't seem to come from any of `-Wall`, `-Wextra` or `-Wpedantic`. Also, when I add this option, I also get a failure

```
  CC    src/asn1_helpers.c
src/asn1_helpers.c:30:5: error: no previous prototype for 'mbedtls_test_asn1_skip_integer' [-Werror=missing-prototypes]
   30 | int mbedtls_test_asn1_skip_integer( unsigned char **p, const unsigned char *end,
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

--- a/library/sha256.c
+++ b/library/sha256.c
@@ -357,6 +357,13 @@ static size_t mbedtls_internal_sha256_process_many_a64_crypto(
     return( processed );
 }
 
+#if defined(MBEDTLS_SHA256_USE_A64_CRYPTO_IF_PRESENT)
+/*
+ * This function is for internal use only if we are building both C and A64
+ * versions, otherwise it is renamed to be the public mbedtls_internal_sha256_process()
+ */
+static
+#endif
 int mbedtls_internal_sha256_process_a64_crypto( mbedtls_sha256_context *ctx,
         const unsigned char data[SHA256_BLOCK_SIZE] )
 {
@@ -402,6 +409,13 @@ int mbedtls_internal_sha256_process_a64_crypto( mbedtls_sha256_context *ctx,
         (d) += local.temp1; (h) = local.temp1 + local.temp2;        \
     } while( 0 )
 
+#if defined(MBEDTLS_SHA256_USE_A64_CRYPTO_IF_PRESENT)
+/*
+ * This function is for internal use only if we are building both C and A64
+ * versions, otherwise it is renamed to be the public mbedtls_internal_sha256_process()
+ */
+static
+#endif
 int mbedtls_internal_sha256_process_c( mbedtls_sha256_context *ctx,
                                 const unsigned char data[SHA256_BLOCK_SIZE] )
 {

--- a/library/sha512.c
+++ b/library/sha512.c
@@ -527,6 +527,13 @@ static size_t mbedtls_internal_sha512_process_many_a64_crypto(
     return( processed );
 }
 
+#if defined(MBEDTLS_SHA512_USE_A64_CRYPTO_IF_PRESENT)
+/*
+ * This function is for internal use only if we are building both C and A64
+ * versions, otherwise it is renamed to be the public mbedtls_internal_sha512_process()
+ */
+static
+#endif
 int mbedtls_internal_sha512_process_a64_crypto( mbedtls_sha512_context *ctx,
                               const unsigned char data[SHA512_BLOCK_SIZE] )
 {
@@ -545,6 +552,13 @@ int mbedtls_internal_sha512_process_a64_crypto( mbedtls_sha512_context *ctx,
 
 #if !defined(MBEDTLS_SHA512_PROCESS_ALT) && !defined(MBEDTLS_SHA512_USE_A64_CRYPTO_ONLY)
 
+#if defined(MBEDTLS_SHA512_USE_A64_CRYPTO_IF_PRESENT)
+/*
+ * This function is for internal use only if we are building both C and A64
+ * versions, otherwise it is renamed to be the public mbedtls_internal_sha512_process()
+ */
+static
+#endif
 int mbedtls_internal_sha512_process_c( mbedtls_sha512_context *ctx,
                                        const unsigned char data[SHA512_BLOCK_SIZE] )
 {
@@ -676,7 +690,7 @@ static size_t mbedtls_internal_sha512_process_many_c(
 
 #if defined(MBEDTLS_SHA512_USE_A64_CRYPTO_IF_PRESENT)
 
-int mbedtls_a64_crypto_sha512_has_support( void )
+static int mbedtls_a64_crypto_sha512_has_support( void )
 {
     static int done = 0;
     static int supported = 0;

